x-pack/plugins/security_solution/server/endpoint/routes/trusted_apps/trusted_apps.test.ts

```diff
@@ -138,24 +232,26 @@ describe('when invoking endpoint trusted apps route handlers', () => {
 
   describe('when creating a trusted app', () => {
     let routeHandler: RequestHandler<undefined, PostTrustedAppCreateRequest>;
-    const createNewTrustedAppBody = (): PostTrustedAppCreateRequest => ({
+    const createNewTrustedAppBody = (): {
+      -readonly [k in keyof PostTrustedAppCreateRequest]: PostTrustedAppCreateRequest[k];
```

Comment:
what does this syntax mean?

Reply:
It makes the object "own keys" mutable (you can mutate them). Its short of writing a `Mutable<>` generic interface (to undo what our `Immutable<>` generic does :)